index.js

```diff
@@ -3,126 +3,222 @@ const express = require('express'),
   morgan = require('morgan'),
   bodyParser = require('body-parser'),
   uuid = require('uuid');
+  const mongoose = require('mongoose');
+  const Models = require('./models.js');
 
 const app = express();
+const Movies = Models.Movie;
+const Users = Models.User;
+mongoose.connect('mongodb://localhost:27017/myFlixDatabase', {
+  useNewUrlParser: true, // Connect the DB (to perform CRUD)
+  useUnifiedTopology: true
+});
 
 app.use(morgan('common'));
-app.use(bodyParser.json());
+app.use(express.json());
 
 // GET requests
 app.get('/', (req, res) => {
   res.send('Welcome to myFlix!');
 });
 
-// CREATE a new User
-app.post('/users', (req, res) => {
-  const newUser = req.body;
-
-  if (newUser.name) {
-    newUser.id = uuid.v4();
-    users.push(newUser);
-    res.status(201).json(newUser)
-  } else {
-    res.status(400).send('New user needs a name.')
-  }
+// Read/ Return a list of all movies
+app.get('/movies', (req, res) => {
+  Movies.find()
+    .then((movies) => {
+      res.status(200).json(movies);
+    })
+    .catch((err) => {
+      console.error(err);
+      res.status(500).send('Error: ' + err);
+    });
```

Comment:
@Thom187 We could do something like this. It was just a suggestion
```
app.get("/movies", async (req, res) => {
  try {
    const movies = await Movies.find();
    if (movies) {
      res.status(200).json(movies);
    } else {
      res.status(400).json({ message: "movies not found" });
    }
  } catch (error) {
    res.status(500).send("Error: " + error);
  }
});
```

Reply:
OK, for now that works for all my GET requests.

I also tried to replace the POST a new user.
Looking for an existing user and create a new one works well but i do not get the new created JSON object as response.
How can I solve this?